phpBB/includes/acp/acp_bbcodes.php

```diff
@@ -172,14 +172,25 @@ function main($id, $mode)
 				);
 				extract($phpbb_dispatcher->trigger_event('core.acp_bbcodes_modify_create', compact($vars)));
 
-				$warn_text = preg_match('%<[^>]*\{text[\d]*\}[^>]*>%i', $bbcode_tpl);
+				$acp_utils   = $phpbb_container->get('text_formatter.acp_utils');
+				$bbcode_info = $acp_utils->analyse_bbcode($bbcode_match, $bbcode_tpl);
+				$warn_unsafe = ($bbcode_info['status'] === 'unsafe');
```

Comment:
I'm not the biggest fan of using strings for the status unless explicitly need. What do you think about adding constants to your interface instead and using these?

Reply:
I don't think it really offers any benefits but if you think it's worth it please feel free to add it to the interface directly.

Reply:
Actually, check out the latest commit and modify as desired. Without an enum type I don't think it really matters what the constants are but if the goal is to program defensively then I think it's a good idea to rework the conditionals as in d4d93160a7e40cd91b0816dcf52ed8365e1d8864. Instead of emitting an error if the status shows it's invalid/unsafe, it throws an error if the status is not valid.